runtime/onert/backend/train/ops/PoolLayer.cc

```diff
@@ -25,7 +31,111 @@ namespace train
 namespace ops
 {
 
-PoolLayer::PoolLayer() : cpu::ops::PoolLayer()
+namespace
+{
+
+// Convert train op type to inter(cpu) op type
+cpu::ops::PoolType convert_to_infer(const train::ops::PoolType &pool_type)
+{
+  switch (pool_type)
+  {
+    case train::ops::PoolType::kMax:
+      return cpu::ops::PoolType::kMax;
+    default:
+      throw std::runtime_error("PoolLayer: Unsupported pool type yet");
+  }
+}
+
+/**
+ * MaxPool Training Kernel Registry
+ *
+ *  - provide forward, bacward interface
+ *  - have addtioinal buffer needed to do backpropagate
+ */
+class MaxPool2D final : public TrainingKernelRegistry
```

Comment:
This implements maxpool2d forward, and backward interface to call the proper kernel. 
Also, it has additional memory used to record the max-argument's index.